src/webgpu/util/f32_interval.ts

```diff
@@ -115,25 +103,28 @@ export interface PointToInterval {
 }
 
 /** Operation used to implement a PointToInterval */
-interface PointToIntervalOp {
+export interface PointToIntervalOp {
   /** @returns acceptance interval for a function at point x */
-  impl: (x: number) => F32Interval;
+  impl: PointToInterval;
 
   /**
    * Calculates where in the domain defined by x the min/max extrema of impl
    * occur and returns a span of those points to be used as the domain instead.
    *
-   * If not defined the ends of the existing domain are assumed to be the
-   * extrema.
+   * If the result is not defined the ends of the existing domain are assumed to
+   * be the extrema.
    *
-   * This is only implemented for functions that meet all of the following
+   * This is only implemented for subclasses that meet all of the following
    * criteria:
    *   a) non-monotonic
    *   b) used in inherited accuracy calculations
    *   c) need to take in an interval for b)
    *      i.e. fooInterval takes in x: number | F32Interval, not x: number
    */
   extrema?: (x: F32Interval) => F32Interval;
+
+  /** @returns an additional interval that inputs need to be within. If not defined, they are only tested for being finite. */
```

Comment:
Why is this external to the implementation? Couldn't it be part of the implementation of `impl` ? Seems fairly easy to create a helper to return a `PointToIntervalOp` that enforces this interval check before calling into the inner `PointToIntervalOp`.

Reply:
Thanks for the clarification, I was _overthinking this_ :tm:, trying to come up with a solution that rolled in the general must be finite requirement, whereas you were talking about just the specific cases that have tighter requirements.

I will implement this change and update the PR.

Reply:
Done

Reply:
👍🏼